src/cdk-experimental/testing/protractor.ts

```diff
@@ -0,0 +1,204 @@
+  options?: Options): Promise<T|null>;
+
+export async function load<T extends ComponentHarness>(
+  componentHarness: ComponentHarnessType<T>, rootSelector = 'body',
+  options?: Options): Promise<T|null> {
+  const root = await getElement(rootSelector, undefined, options);
+  if (root === null) {
+    return null;
+  }
+  const locator = new ProtractorLocator(root);
+  return new componentHarness(locator);
+}
+
+/**
+ * Gets the corresponding ElementFinder for the root of a TestElement.
+ */
+export function getElementFinder(testElement: TestElement): ElementFinder {
+  if (testElement instanceof ProtractorElement) {
+    return testElement.element;
+  }
+
+  throw new Error('Invalid element provided');
+}
+
+class ProtractorLocator implements Locator {
+  private _root: ProtractorElement;
+
+  constructor(private _rootFinder: ElementFinder) {
+    this._root = new ProtractorElement(this._rootFinder);
+  }
+
+  host(): TestElement {
+    return this._root;
+  }
+
+  async find(css: string, options?: Options): Promise<TestElement|null> {
+    const finder = await getElement(css, this._rootFinder, options);
+    if (finder === null) {
+      return null;
+    }
+    return new ProtractorElement(finder);
+  }
+
+  async findAll(css: string): Promise<TestElement[]> {
+    const elementFinders = this._rootFinder.all(by.css(css));
+    const res: TestElement[] = [];
+    await elementFinders.each(el => {
+      if (el) {
+        res.push(new ProtractorElement(el));
+      }
+    });
+    return res;
+  }
+
+  async load<T extends ComponentHarness>(
+    componentHarness: ComponentHarnessType<T>, css: string,
+    options?: Options): Promise<T|null> {
+    const root = await getElement(css, this._rootFinder, options);
+    if (root === null) {
+      return null;
+    }
+    const locator = new ProtractorLocator(root);
+    return new componentHarness(locator);
+  }
+
+  async loadAll<T extends ComponentHarness>(
+    componentHarness: ComponentHarnessType<T>,
+    rootSelector: string,
+  ): Promise<T[]> {
+    const roots = this._rootFinder.all(by.css(rootSelector));
+    const res: T[] = [];
+    await roots.each(el => {
+      if (el) {
+        const locator = new ProtractorLocator(el);
+        res.push(new componentHarness(locator));
+      }
+    });
+    return res;
+  }
+}
+
+class ProtractorElement implements TestElement {
+  constructor(readonly element: ElementFinder) {}
+
+  blur(): Promise<void> {
+    return toPromise<void>(this.element['blur']());
+  }
+
+  clear(): Promise<void> {
+    return toPromise<void>(this.element.clear());
+  }
+
+  click(): Promise<void> {
+    return toPromise<void>(this.element.click());
+  }
+
+  focus(): Promise<void> {
+    return toPromise<void>(this.element['focus']());
+  }
+
+  getCssValue(property: string): Promise<string> {
+    return toPromise<string>(this.element.getCssValue(property));
+  }
+
+  async hover(): Promise<void> {
```

Comment:
How would we handle this if the consumer was running the test against a touch screen device?

Reply:
I don't think it would be an issue necessarily. Protractor must have some way to handle `mouseMove` on touch devices already, even if its to just do nothing. It would make sense to add a `tap` action as well. I can do that in a follow-up PR.

Reply:
AFAIK Protractor simulates real user input at the OS level so I'm not sure how it work. My main concern with `hover` was for unit tests and it should be a bit easier to simulate it there though.